src/components/common/RestEntity/README.md

```diff
@@ -0,0 +1,882 @@
+
+LOADING:
+
+```javascript
+// if data is still being fetched
+// we'll show the user that we're loading up their data
+return <div>Loading...</div>;
+```
+
+ERROR:
+
+```javascript
+// if we hit an error, then we'll render this warning to our user!
+return <p>Oh no! There's been a problem!</p>;
+```
+
+SUCCESS:
+
+```javascript
+// if data was successfully fetched, 
+// then we'll render our program JSX to display that data
+return  (
+  <div>
+    <h1>Program Name: {data.programName}</h1>
+    <p>Description: {data.description}</p>
+  </div>
+);
+```
+
+Do you notice anything interesting there? Of the three different states, only ONE depends on external data of any sort—SUCCESS. We can display anything we want to for a Loading message or an Error message... the only thing that depends on the actual data is the success!! And logically so, because if you're loading data or failed to get the data, then you definitely don't have any data that you could depend on.
+
+So let's plug our `Error` and `Loading` states into our new and improved `ProgramComponent.js`
+
+```javascript
+import React from "react";
+import RestEntity from "./RestEntity";
+
+const ProgramComponent = props => {
+  return (
+    <RestEntity href={`/programs/program/${props.programId}`}>
+
+      {/* WE STILL DON'T KNOW WHAT TO DO HERE */}
+      <RestEntity.Singleton />
+      
+      {/* The JSX inside this will only display if we hit an error */}
+      <RestEntity.Error>
+        <p>Oh no! There's been a problem!</p>
+      </RestEntity.Error>
+      
+      {/* The JSX inside this will only display when we're loading up data */}
+      <RestEntity.Loading>
+        <div>Loading...</div>
+      </RestEntity.Loading>
+
+    </RestEntity>
+  );
+};
+
+export default ProgramComponent;
+```
+
+Wow! We now have a component that can:
+
+- Fetch Data from the endpoint that we specified
+
+- Display a Loading message when that data is still being fetched.
+
+- Display a warning message if something goes wrong while fetching the data
+
+There's just one piece missing... and it's the most important piece: HOW do we display the JSX that presents the data we care about.
+
+In this example, we still need to render this on SUCCESS:
+
+```javascript
+// if data was successfully fetched, 
+// then we'll render our program JSX to display that data
+(
+  <div>
+    <h1>Program Name: {data.programName}</h1>
+    <p>Description: {data.description}</p>
+  </div>
+);
+```
+
+That's where the `RestEntity.Singleton` sub-component comes in. The most important prop that `Singleton` receives is the `props.component`. The value assigned to this prop can look two different ways:
+
+- A "callback Component"
+- A straight up Component
+
+<br>
+<details>
+<summary>Quick Refresher on Callbacks & Functions</summary>
+
+---
+
+### A Quick refresher on callbacks & functions
+
+---
+
+#### Functions First
+
+- functions can be ***arrow*** functions or ***traditional (non-arrow)***
```

Comment:
This supporting info could be all gathered in one place or left for external links, etc.